metadata-ingestion/src/datahub/ingestion/source/dbt.py

```diff
@@ -103,7 +109,7 @@ def report_stale_entity_soft_deleted(self, urn: str) -> None:
         self.soft_deleted_stale_entities.append(urn)
 
 
-class DBTConfig(StatefulIngestionConfigBase):
+class DBTConfig(AwsSourceConfig, StatefulIngestionConfigBase):
```

Comment:
Unfortunately `AwsSourceConfig` seems to bring in a bunch of stuff you don't really need to pull in. e.g. `database_pattern`, `table_pattern`, `env` etc. 

I suggest you refactor AwsSourceConfig into AwsConnectionConfig to extract that code out, and pull that in as an optional field into DBTConfig.